plugins/ScheduledReports/tests/Fixtures/ReportSubscription.php

```diff
@@ -7,37 +7,222 @@
+                    if ($pluginName != 'ExampleAPI' && $methodName != 'xssReportforTwig' && $methodName != 'xssReportforAngular') {
+                        return;
+                    }
+
+                    if (!empty($_GET['forceError']) || !empty($_POST['forceError'])) {
+                        throw new \Exception("forced exception");
+                    }
+
+                    $dataTable = new DataTable();
+                    $dataTable->addRowFromSimpleArray([
+                        'label' => $this->angularXssLabel,
+                        'nb_visits' => 10,
+                    ]);
+                    $dataTable->addRowFromSimpleArray([
+                        'label' => $this->twigXssLabel,
+                        'nb_visits' => 15,
+                    ]);
+                    $result = $dataTable;
+                }],
+            ]),
         ];
+    }
+}
+
+class XssReport extends Report
+{
+    private $xssType;
+
+    protected function init()
+    {
+        parent::init();
+
+        $this->metrics        = array('nb_visits');
+        $this->order = 10;
+
+        $action = Common::getRequestVar('actionToWidgetize', false) ?: Common::getRequestVar('action', false);
+        if ($action == 'xssReportforTwig') {
+            $this->initForXss('forTwig');
+        } else if ($action == 'xssReportforAngular') {
+            $this->initForXss('forAngular');
+        }
+    }
+
+    public function initForXss($type)
+    {
+        $this->xssType = $type;
+
+        $xssTesting = new XssTesting();
+        $this->dimension      = new XssDimension();
+        $this->dimension->initForXss($type);
+        $this->name           = $xssTesting->$type('reportname');
+        $this->documentation  = $xssTesting->$type('reportdoc');
+        $this->categoryId = $xssTesting->$type('category');
+        $this->subcategoryId = $xssTesting->$type('subcategory');
+        $this->processedMetrics = [new XssProcessedMetric($type)];
+        $this->module = 'ExampleAPI';
+        $this->action = 'xssReport' . $type;
+        $this->id = 'ExampleAPI.xssReport' . $type;
+    }
+}
+
+class XssDimension extends VisitDimension
+{
+    public $type = Dimension::TYPE_NUMBER;
 
-        API::getInstance()->addReport($this->idSite, 'description', 'day', 3, $reportType, $reportFormat, $reports, $parameters);
+    private $xssType;
 
-        Db::query("INSERT INTO " . Common::prefixTable('report_subscriptions') . "(idreport, token, email) VALUES (1, 'mycustomtoken', 'any@matomo.org')");
+    public function initForXss($type)
+    {
+        $xssTesting = new XssTesting();
+
+        $this->xssType = $type;
+        $this->nameSingular = $xssTesting->$type('dimensionname');
+        $this->columnName = 'xsstestdim';
+        $this->category = $xssTesting->$type('category');
+    }
+
+    public function getId()
+    {
+        return 'XssTest.XssDimension.' . $this->xssType;
+    }
+}
+
+class XssProcessedMetric extends ProcessedMetric
```

Comment:
Is it needed to copy this code or can we do without maybe? or reuse the other fixture?